cartoframes/dataset.py

```diff
@@ -24,51 +33,74 @@ class Dataset(object):
     PUBLIC = 'public'
     LINK = 'link'
 
+    TABLE_TYPE = 'Table'
+    QUERY_TYPE = 'Query'
+    DATAFRAME_TYPE = 'DataFrame'
+    GEODATAFRAME_TYPE = 'GeoDataFrame'
+
     DEFAULT_RETRY_TIMES = 3
 
-    def __init__(self, carto_context, table_name, schema='public', df=None):
-        self.cc = carto_context
-        self.table_name = normalize_name(table_name)
+    def __init__(self, type, data, context=None, schema='public', table_name=None):
```

Comment:
I agree with @oleurud's comments. I don't have a good proposal, but I know that it feels burdensome to type `Dataset('geodataframe', gdf)`.

We should be careful to not use `type` because it's a builtin function

Reply:
Agree with both! I thought that the final Dataset API was the `from_x` methods instead of calling directly the constructor. I did the changes just to complete the integration and pass all the tests, but we need to define the possible interfaces for Dataset and then update the implementation.